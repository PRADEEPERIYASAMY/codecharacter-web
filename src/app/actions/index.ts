export * from 'app/actions/dashboard';
