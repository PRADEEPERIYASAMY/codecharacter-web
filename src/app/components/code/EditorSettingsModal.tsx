import { faTimes } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { themes } from 'app/components/code/Editor'
import * as styles from 'app/styles/Modal.module.css';
import classnames from 'classnames';
import * as React from 'react';
import { Button, Col, FormGroup, Grid, Row } from 'react-bootstrap';


// tslint:disable-next-line:variable-name
export class EditorSettingsModal extends React.Component<EditorSettingsModal.Props, {}> {

  public render() {
    const {
      showCustomizationModal,
      toggleCustomizationPanel,
      fontSize,
      changeFontSize,
      theme,
      changeTheme
    } = this.props
    const fontSizeOptions = []
    for (let i = 8; i <= 40; i += 2) {
      fontSizeOptions.push(i)
    }
    return (
      <Grid
        fluid={true}
        className={classnames(styles.Modal)}
        style={{
          opacity: showCustomizationModal ? 1 : 0,
          transform: showCustomizationModal ? 'translateY(0)' : 'translateY(-100vh)',
        }}
      >
        <Row className="justify-content-between py-2 pl-3">
          <Col className="text-light my-auto">SETTINGS</Col>
          <Col>
            <Button
              bsSize="xsmall"
              onClick={() => toggleCustomizationPanel(!showCustomizationModal)}
            >
              <FontAwesomeIcon icon={faTimes} color={'white'} />
            </Button>
          </Col>
        </Row>
        <Row>
          <Col sm={12} className={classnames('mb-1', styles.setting)}>
            <div className="font-weight-bold py-2">Font Size</div>
            <p>The size of text in the code editor</p>
            <FormGroup controlId="fontSize">
              <select
                className={classnames('form-control', styles.formControlSelect)}
                value={fontSize}
                onChange={(e) => changeFontSize(Number(e.target.value))}
              >
                {
                  fontSizeOptions.map((font: number) => (
                    <option value={font} key={font}>{font}</option>
                  ))
                }
              </select>
            </FormGroup>
          </Col>
          <Col sm={12} className={classnames('mb-1', styles.setting)}>
            <div className="font-weight-bold py-2">Editor Theme</div>
            <p>The syntax and overall theme of the code editor</p>
            <FormGroup controlId="theme">
              <select
                className={classnames('form-control', styles.formControlSelect)}
                value={theme}
                onChange={(e) => changeTheme(e.target.value)}
              >
                {themes.map((themeValue: string) => (
                  <option value={themeValue} key={themeValue}>{themeValue}</option>
                ))}
              </select>
            </FormGroup>
          </Col>
        </Row>
      </Grid>
    );
  }
};

export namespace EditorSettingsModal {
  export interface StateProps {
    fontSize: number;
    theme: string;
    showCustomizationModal: boolean;
  }

  export interface DispatchProps {
    changeFontSize: (fontSize: number) => void;
    changeTheme: (theme: string) => void;
    toggleCustomizationPanel: (showCustomizationPanel: boolean) => void;
  }

  export type Props = {} & StateProps & DispatchProps;
}
