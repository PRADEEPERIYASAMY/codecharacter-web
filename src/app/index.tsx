import { Dashboard } from 'app/components/Dashboard';
import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router';

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <Switch>
    <Route path="/" component={Dashboard} />
  </Switch>
));
