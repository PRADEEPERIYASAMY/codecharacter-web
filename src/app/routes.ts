export enum Routes {
  ROOT = '/',
  LOGIN = '/login',
  REGISTER = '/register',
  LEADERBOARD = '/leaderboard',
  USER_PROFILE_MODEL = '/profile',
<<<<<<< HEAD
  GITHUB_OAUTH = '/login/github',
  GOOGLE_OAUTH = '/login/google',
  USER_ACTIVATION = '/user-activate',
=======
  PROFILE_USER_STATS = '/profile/:username',
>>>>>>> Add route for /profile/:user and Add the required container
}
