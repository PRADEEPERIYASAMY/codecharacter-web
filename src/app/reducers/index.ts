import { codeReducer } from 'app/reducers/code/Code';
import { editorReducer } from 'app/reducers/code/Editor';
import { dashboardReducer } from 'app/reducers/Dashboard';
import { gameLogReducer } from 'app/reducers/GameLog';
import { leaderboardReducer } from 'app/reducers/Leaderboard';
import { notificationReducer } from 'app/reducers/Notification';
import { userReducer } from 'app/reducers/User';
import * as CodeInterfaces from 'app/types/code/Code';
import * as EditorInterfaces from 'app/types/code/Editor';
import * as DashboardInterfaces from 'app/types/Dashboard';
import * as GameLogInterfaces from 'app/types/GameLog';
import * as LeaderboardInterfaces from 'app/types/Leaderboard';
import * as UserInterfaces from 'app/types/User';
import { routerReducer, RouterState } from 'react-router-redux';
import { combineReducers } from 'redux';

export const rootReducer = combineReducers({
  code: codeReducer,
  dashboard: dashboardReducer,
  editor: editorReducer,
  gameLog: gameLogReducer,
  leaderboard: leaderboardReducer,
  notification: notificationReducer,
  router: routerReducer,
  user: userReducer,
});

export interface RootState {
  code: CodeInterfaces.CodeStoreState;
  dashboard: DashboardInterfaces.DashboardStoreState;
  editor: EditorInterfaces.EditorStoreState;
  leaderboard: LeaderboardInterfaces.LeaderboardStoreState;
  router: RouterState;
  gameLog: GameLogInterfaces.GameLogStoreState;
  user: UserInterfaces.UserStoreState;
}
