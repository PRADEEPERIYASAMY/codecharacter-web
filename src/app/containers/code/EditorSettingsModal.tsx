import { DashboardActions, EditorActions } from 'app/actions';
import { EditorSettingsModal } from 'app/components/code/EditorSettingsModal';
import { RootState } from 'app/reducers';
import { connect } from 'react-redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    fontSize: rootState.editor.editorOptions.fontSize,
    showEditorSettingsPanel: rootState.dashboard.showEditorSettingsPanel,
    theme: rootState.editor.editorOptions.theme,
  };
};

const editorSettingsModalContainer = connect<
  EditorSettingsModal.StateProps,
  EditorSettingsModal.DispatchProps,
  {}
>(
  mapStateToProps,
  {
    changeFontSize: EditorActions.changeFontSize,
    changeTheme: EditorActions.changeTheme,
    toggleEditorSettingsPanel: DashboardActions.toggleEditorSettingsPanel,
  },
)(EditorSettingsModal);

export default editorSettingsModalContainer;
