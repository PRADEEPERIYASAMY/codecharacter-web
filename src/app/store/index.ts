import { logger } from 'app/middleware';
import { rootReducer } from 'app/reducers';
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

export function configureStore(initialState?: object) {
  let middleware = applyMiddleware(logger);

  if (process.env.NODE_ENV !== 'production') {
    middleware = composeWithDevTools(middleware);
  }

  const store = createStore(rootReducer, initialState, middleware);

  if (module.hot) {
    module.hot.accept('app/reducers', () => {
      const nextReducer = require('app/reducers');
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}
