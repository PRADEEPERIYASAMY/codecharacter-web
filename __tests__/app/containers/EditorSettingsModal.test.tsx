import EditorSettingsModal from 'app/containers/code/EditorSettingsModal';
import { configureStore } from 'app/store';
import { mount } from 'enzyme';
import * as React from 'react';
// import { RootState } from 'app/reducers';

describe ('EditorSettingsModal Container', () => {
    const store = configureStore();

    it ('Should render', () => {
        const wrapper = mount(<EditorSettingsModal/>, {
            context: {
                store,
            }
        });
        expect(wrapper).toMatchSnapshot();
    });

    it('Should change font', () => {
        const wrapper = mount(<EditorSettingsModal />, {
            context: {
                store,
            }
        });

        expect(wrapper.find('select.font-size-control').props().value).toBe(16);
        wrapper.find('select.font-size-control').simulate('change', { target: { value: '32' } });
        expect(wrapper.find('select.font-size-control').props().value).toBe(32);
    });

    it('Should change theme', () => {
        const wrapper = mount(<EditorSettingsModal />, {
            context: {
                store,
            }
        });

        expect(wrapper.find('select.theme-control').props().value).toBe('monokai');
        wrapper.find('select.theme-control').simulate('change', { target: { value: 'github' } });
        expect(wrapper.find('select.theme-control').props().value).toBe('github');
    });
});
