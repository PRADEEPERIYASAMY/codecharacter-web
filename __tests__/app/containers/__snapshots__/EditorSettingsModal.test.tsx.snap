// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorSettingsModal Container Should render 1`] = `
<Connect(EditorSettingsModal)>
  <EditorSettingsModal
    changeFontSize={[Function]}
    changeTheme={[Function]}
    fontSize={16}
    showCustomizationModal={false}
    theme="monokai"
    toggleCustomizationPanel={[Function]}
  >
    <Grid
      bsClass="container"
      className=""
      componentClass="div"
      fluid={true}
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(-100vh)",
        }
      }
    >
      <div
        className="container-fluid"
        style={
          Object {
            "opacity": 0,
            "transform": "translateY(-100vh)",
          }
        }
      >
        <Row
          bsClass="row"
          className="justify-content-between py-2 pl-3"
          componentClass="div"
        >
          <div
            className="justify-content-between py-2 pl-3 row"
          >
            <Col
              bsClass="col"
              className="text-light my-auto"
              componentClass="div"
            >
              <div
                className="text-light my-auto"
              >
                SETTINGS
              </div>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
            >
              <div
                className=""
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsSize="xsmall"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-xs btn-default"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <FontAwesomeIcon
                      border={false}
                      className=""
                      color="white"
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            352,
                            512,
                            Array [],
                            "f00d",
                            "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                          ],
                          "iconName": "times",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-times fa-w-11 "
                        color="white"
                        data-icon="times"
                        data-prefix="fas"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 352 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                  </button>
                </Button>
              </div>
            </Col>
          </div>
        </Row>
        <Row
          bsClass="row"
          componentClass="div"
        >
          <div
            className="row"
          >
            <Col
              bsClass="col"
              className="mb-1"
              componentClass="div"
              sm={12}
            >
              <div
                className="mb-1 col-sm-12"
              >
                <div
                  className="font-weight-bold py-2"
                >
                  Font Size
                </div>
                <p>
                  The size of text in the code editor
                </p>
                <FormGroup
                  bsClass="form-group"
                  controlId="fontSize"
                >
                  <div
                    className="form-group"
                  >
                    <select
                      className="form-control font-size-control"
                      onChange={[Function]}
                      value={16}
                    >
                      <option
                        key="8"
                        value={8}
                      >
                        8
                      </option>
                      <option
                        key="10"
                        value={10}
                      >
                        10
                      </option>
                      <option
                        key="12"
                        value={12}
                      >
                        12
                      </option>
                      <option
                        key="14"
                        value={14}
                      >
                        14
                      </option>
                      <option
                        key="16"
                        value={16}
                      >
                        16
                      </option>
                      <option
                        key="18"
                        value={18}
                      >
                        18
                      </option>
                      <option
                        key="20"
                        value={20}
                      >
                        20
                      </option>
                      <option
                        key="22"
                        value={22}
                      >
                        22
                      </option>
                      <option
                        key="24"
                        value={24}
                      >
                        24
                      </option>
                      <option
                        key="26"
                        value={26}
                      >
                        26
                      </option>
                      <option
                        key="28"
                        value={28}
                      >
                        28
                      </option>
                      <option
                        key="30"
                        value={30}
                      >
                        30
                      </option>
                      <option
                        key="32"
                        value={32}
                      >
                        32
                      </option>
                      <option
                        key="34"
                        value={34}
                      >
                        34
                      </option>
                      <option
                        key="36"
                        value={36}
                      >
                        36
                      </option>
                      <option
                        key="38"
                        value={38}
                      >
                        38
                      </option>
                      <option
                        key="40"
                        value={40}
                      >
                        40
                      </option>
                    </select>
                  </div>
                </FormGroup>
              </div>
            </Col>
            <Col
              bsClass="col"
              className="mb-1"
              componentClass="div"
              sm={12}
            >
              <div
                className="mb-1 col-sm-12"
              >
                <div
                  className="font-weight-bold py-2"
                >
                  Editor Theme
                </div>
                <p>
                  The syntax and overall theme of the code editor
                </p>
                <FormGroup
                  bsClass="form-group"
                  controlId="theme"
                >
                  <div
                    className="form-group"
                  >
                    <select
                      className="form-control theme-control"
                      onChange={[Function]}
                      value="monokai"
                    >
                      <option
                        key="monokai"
                        value="monokai"
                      >
                        monokai
                      </option>
                      <option
                        key="github"
                        value="github"
                      >
                        github
                      </option>
                      <option
                        key="tomorrow"
                        value="tomorrow"
                      >
                        tomorrow
                      </option>
                      <option
                        key="kuroir"
                        value="kuroir"
                      >
                        kuroir
                      </option>
                      <option
                        key="twilight"
                        value="twilight"
                      >
                        twilight
                      </option>
                      <option
                        key="xcode"
                        value="xcode"
                      >
                        xcode
                      </option>
                      <option
                        key="textmate"
                        value="textmate"
                      >
                        textmate
                      </option>
                      <option
                        key="solarized_dark"
                        value="solarized_dark"
                      >
                        solarized_dark
                      </option>
                      <option
                        key="solarized_light"
                        value="solarized_light"
                      >
                        solarized_light
                      </option>
                      <option
                        key="terminal"
                        value="terminal"
                      >
                        terminal
                      </option>
                    </select>
                  </div>
                </FormGroup>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Grid>
  </EditorSettingsModal>
</Connect(EditorSettingsModal)>
`;
